recipes/mongo-cxx-driver/all/conanfile.py

```diff
@@ -47,9 +47,9 @@ def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("mongo-c-driver/1.23.2")
+        self.requires("mongo-c-driver/1.23.5", transitive_headers=True)
```

Comment:
I found boost headers in the public headers of the lib, but not any ``mongo-c`` stuff in the public headers. Are you sure they should be transitive?

Reply:
@memsharded 
Thank you for your review.
It's my mistake.
mongo-c-driver's headers are included in test_package.cpp.
But the symbols in it's headers are not used.

I modify test_package.cpp.